### PR TITLE
Numpy roll large tensor fix fix

### PR DESCRIPTION
**Pull request by Zha0q1**
This pr adds large tensor support to `np.roll`

local run:
```
ubuntu@ip-172-31-38-169:~/incubator-mxnet$ pytest tests/nightly/test_np_large_array.py::test_roll
/home/ubuntu/anaconda3/lib/python3.7/importlib/_bootstrap.py:219: RuntimeWarning: numpy.ufunc size changed, may indicate binary incompatibility. Expected 192 from C header, got 216 from PyObject
  return f(*args, **kwds)
/home/ubuntu/anaconda3/lib/python3.7/importlib/_bootstrap.py:219: RuntimeWarning: numpy.ufunc size changed, may indicate binary incompatibility. Expected 192 from C header, got 216 from PyObject
  return f(*args, **kwds)
/home/ubuntu/anaconda3/lib/python3.7/importlib/_bootstrap.py:219: RuntimeWarning: numpy.ufunc size changed, may indicate binary incompatibility. Expected 192 from C header, got 216 from PyObject
  return f(*args, **kwds)
================================================ test session starts ================================================
platform linux -- Python 3.7.7, pytest-5.4.1, py-1.8.1, pluggy-0.13.1
rootdir: /home/ubuntu/incubator-mxnet, inifile: pytest.ini
plugins: remotedata-0.3.2, openfiles-0.4.0, astropy-header-0.1.2, hypothesis-5.8.3, arraydiff-0.3, doctestplus-0.5.0
collected 1 item                                                                                                    

tests/nightly/test_np_large_array.py   .                                                                        [100%]

================================================ 1 passed in 24.10s =================================================
```

### CODE REVIEW

**Comment by Zha0q1** [comment]
@mxnet-bot run ci [sanity]

**Review by access2rohit — APPROVED** [review]

**Comment by Zha0q1** [comment]
@mxnet-bot run ci [centos-gpu, unix-cpu]

**Comment by Zha0q1** [comment]
@samskalicky would you review and merge?

**Comment by Zha0q1** [comment]
@mseth10

File: src/operator/numpy/np_matrix_op-inl.h [hunk 1]
```diff
@@ -487,12 +487,12 @@ void NumpyRollCompute(const nnvm::NodeAttrs& attrs,
   CHECK_EQ(req.size(), 1U);
   if (inputs[0].Size() == 0U) return;
   const NumpyRollParam& param = nnvm::get<NumpyRollParam>(attrs.parsed);
-  const index_t ndim(inputs[0].shape_.ndim());
+  const int ndim(inputs[0].shape_.ndim());
```
Comment (samskalicky) [inline src/operator/numpy/np_matrix_op-inl.h:490]: so for LTS the number of dimensions in the array is still (2^31)-1, but the size of a dimension can be bigger than 2^32?

Comment (Zha0q1) [reply src/operator/numpy/np_matrix_op-inl.h:490]: Yes. Number of dims has always been int. Here the original author used index_t but there is no point in doing so. Plus I think we have a hard limit for any tensors to be <= 10 dims

Comment (Zha0q1) [reply src/operator/numpy/np_matrix_op-inl.h:490]: size of individual dims and total size can both be >= 2**31 when large tensor is on

**Review by samskalicky — APPROVED** [review]
Thanks for the explanation

**Pull request merged**